Objective-C/Tests/ReplicatorTest+PendingDocIds.m

```diff
@@ -32,82 +35,196 @@ @implementation ReplicatorTest_PendingDocIds
 
 #pragma mark - Helper Methods
 
-- (void) createDocs: (int)count action: (NSString*)action {
-    for (int i = 0; i < count; i++) {
+- (NSSet*) createDocs {
+    NSMutableSet<NSString*>* docIds = [NSMutableSet set];
+    for (int i = 0; i < kNoOfDocument; i++) {
         NSString* docId = [NSString stringWithFormat: kDocIdFormat, i];
         CBLMutableDocument* doc = [self createDocument: docId];
-        [doc setString: action forKey: kActionKey];
+        [doc setString: kCreateActionValue forKey: kActionKey];
         [self saveDocument: doc];
+        [docIds addObject: docId];
     }
+    return [NSSet setWithSet: docIds];
 }
 
-- (void) updateDocs: (int)count action: (NSString*)action {
-    for (int i = 0; i < count; i++) {
-        NSString* docId = [NSString stringWithFormat: kDocIdFormat, i];
+- (void) validatePendingDocumentIds: (NSSet*)docIds count: (NSUInteger)count {
+    id target = [[CBLDatabaseEndpoint alloc] initWithDatabase: otherDB];
+    id config = [self configWithTarget: target type: kCBLReplicatorTypePush continuous: NO];
+
+    __block id<CBLListenerToken> token;
+    __block CBLReplicator* replicator;
+    [self run: config reset: NO errorCode: 0 errorDomain: nil onReplicatorReady: ^(CBLReplicator* r) {
```

Comment:
I think we can also check the pendingDocumentIds before starting the replicator as well.

Reply:
That part is not yet implemented, that is done in a separate PR.

Reply:
```suggestion
    [self run: config reset: NO errorCode: 0 errorDomain: nil onReplicatorReady: ^(CBLReplicator* r) {
        // FIXME: Check pending-document-id before starting the replicator. depends on #2569
```